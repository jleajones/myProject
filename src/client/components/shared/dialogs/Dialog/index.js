import React from 'react';
import PropTypes from 'prop-types';
import MuiDialog from '@material-ui/core/Dialog';

const Dialog = ({ onClose, open, label, children, fullScreen }) => {
  return (
    <MuiDialog
      onClose={onClose}
      aria-labelledby={label}
      open={open}
      fullScreen={fullScreen}
    >
      {children}
    </MuiDialog>
  );
};

Dialog.defaultProps = {
  open: false,
  fullScreen: false
};

Dialog.propTypes = {
  onClose: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  open: PropTypes.bool,
  fullScreen: PropTypes.bool
};

export default Dialog;
