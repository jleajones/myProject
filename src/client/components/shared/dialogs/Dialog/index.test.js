import React from 'react';
import { shallow } from 'enzyme';

import Dialog from './index';

describe('<Dialog />', () => {
  test('Renders component without crashing', () => {
    const handleCloseDialog = jest.fn();
    const open = true;
    const title = 'test-dialog';
    const component = shallow(
      <Dialog
        onClose={handleCloseDialog}
        label={title}
        title={title}
        open={open}
      >
        <div>some test content</div>
      </Dialog>
    );

    expect(component).toMatchSnapshot();
  });
});
