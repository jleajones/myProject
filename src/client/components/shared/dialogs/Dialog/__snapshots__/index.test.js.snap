// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> Renders component without crashing 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="test-dialog"
  fullScreen={false}
  onClose={[MockFunction]}
  open={true}
>
  <div>
    some test content
  </div>
</WithStyles(ForwardRef(Dialog))>
`;
